crates/oxc_linter/src/rules/eslint/no_control_regex.rs

```diff
@@ -1,25 +1,88 @@
+use std::fmt::Write;
+
 use itertools::Itertools as _;
+
 use oxc_diagnostics::OxcDiagnostic;
 use oxc_macros::declare_oxc_lint;
 use oxc_regular_expression::{
-    ast::{CapturingGroup, Character, Pattern},
+    ast::{CapturingGroup, Character, CharacterKind, Pattern},
     visit::{Visit, walk},
 };
 use oxc_span::Span;
 
 use crate::{AstNode, context::LintContext, rule::Rule, utils::run_on_regex_node};
 
-fn no_control_regex_diagnostic(count: usize, regex: &str, span: Span) -> OxcDiagnostic {
+fn no_control_regex_diagnostic(control_chars: &[Character], span: Span) -> OxcDiagnostic {
+    let count = control_chars.len();
     debug_assert!(count > 0);
-    let (message, help) = if count == 1 {
-        ("Unexpected control character", format!("'{regex}' is not a valid control character."))
-    } else {
-        ("Unexpected control characters", format!("'{regex}' are not valid control characters."))
-    };
 
-    OxcDiagnostic::warn(message).with_help(help).with_label(span)
-}
+    let mut octal_chars = Vec::new();
+    let mut null_chars = Vec::new();
+    let mut other_chars = Vec::new();
 
+    for ch in control_chars {
+        match ch.kind {
+            CharacterKind::Octal1 | CharacterKind::Octal2 | CharacterKind::Octal3 => {
+                octal_chars.push(ch);
+            }
+            CharacterKind::Null => {
+                null_chars.push(ch);
+            }
+            _ => {
+                other_chars.push(ch);
+            }
+        }
+    }
+
+    let mut help = String::new();
+
+    if !other_chars.is_empty() {
+        let regexes = other_chars.iter().join(", ");
+        writeln!(
+            help,
+            "'{regexes}' {} not {}valid control character{}.",
```

Comment:
BTW I guess this message is wrong. This rule (`no-control-regex`) detects [control characters](https://en.wikipedia.org/wiki/Control_character#In_ASCII), so this message should output "'\x01' is a control character" instead.

Reply:
yeah - you're right, thanks for catching